extension/chrome/settings/modules/contact_page.ts

```diff
@@ -6,76 +6,102 @@ import { Catch } from '../../../js/common/platform/catch.js';
+      }
+    }
+  }
 
-  const renderFields = (result: BackendRes.FcAccountUpdate$result) => {
+  setHandlers() {
+    this.S.cached('action_update').click(this.setHandlerPrevent('double', () => this.onUpdateHandler()));
+    this.S.cached('action_close').click(this.setHandler(() => this.onCloseHandler()));
+  }
+
+  private renderFields(result: BackendRes.FcAccountUpdate$result) {
     if (result.alias) {
       const me = Backend.url('me', result.alias);
       const meEscaped = Xss.escape(me);
       const meEscapedDisplay = Xss.escape(me.replace('https://', ''));
-      Xss.sanitizeRender(S.cached('status'), `Your contact page is currently <b class="good">enabled</b> at <a href="${meEscaped}" target="_blank">${meEscapedDisplay}</a></span>`);
-      S.cached('hide_if_active').css('display', 'none');
-      S.cached('show_if_active').css('display', 'inline-block');
-      S.cached('input_email').val(result.email);
-      S.cached('input_intro').val(result.intro);
-      S.cached('input_alias').val(result.alias);
-      S.cached('input_name').val(result.name);
+      Xss.sanitizeRender(this.S.cached('status'), `Your contact page is currently <b class="good">enabled</b> at <a href="${meEscaped}" target="_blank">${meEscapedDisplay}</a></span>`);
+      this.S.cached('hide_if_active').css('display', 'none');
+      this.S.cached('show_if_active').css('display', 'inline-block');
+      this.S.cached('input_email').val(result.email);
+      this.S.cached('input_intro').val(result.intro);
+      this.S.cached('input_alias').val(result.alias);
+      this.S.cached('input_name').val(result.name);
       if (result.photo) {
-        S.cached('photo').attr('src', result.photo);
+        this.S.cached('photo').attr('src', result.photo);
       }
-      attachJs.initAttDialog('fineuploader', 'select_photo', {
+      this.attachJs.initAttDialog('fineuploader', 'select_photo', {
         attAdded: async file => {
-          newPhotoFile = file;
+          this.newPhotoFile = file;
           Xss.sanitizeReplace('#select_photo', Ui.e('span', { text: file.name }));
         }
       });
     } else {
-      S.cached('management_account').text(result.email).parent().removeClass('display_none');
-      Xss.sanitizeRender(S.cached('status'), 'Your contact page is currently <b class="bad">disabled</b>. <a href="#" class="action_enable">Enable contact page</a>');
-      S.now('action_enable').click(Ui.event.prevent('double', enableContactPage));
+      this.S.cached('management_account').text(result.email).parent().removeClass('display_none');
+      Xss.sanitizeRender(this.S.cached('status'), 'Your contact page is currently <b class="bad">disabled</b>. <a href="#" class="action_enable">Enable contact page</a>');
+      this.S.now('action_enable').click(Ui.event.prevent('double', () => this.enableContactPage()));
```

Comment:
Please use `this.setHandlerPrevent`

Reply:
and then `(this.enableContactPage)` directly

Reply:
Sorry, I missed that!
Thanks!